In [2]:
import pandas as pd
df = pd.read_csv('Sample - Superstore.csv', encoding='latin1')
print("Dataset loaded successfully!")

Dataset loaded successfully!


In [3]:
print(df.head())
print(df.tail())
print(df.shape)
print(df.columns.tolist())
print(df.dtypes)

   Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   Sout

In [9]:
print(df.isnull().sum())

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64


In [10]:
df.fillna('Unknown', inplace=True)
print(df.isnull().sum())

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64


In [ ]:

high_sales = df[df['Sales'] > 1000]
print(high_sales.shape)


print(df[['Order ID', 'Category', 'Sales', 'Quantity']].head())

(468, 21)
         Order ID         Category     Sales  Quantity
0  CA-2016-152156        Furniture  261.9600         2
1  CA-2016-152156        Furniture  731.9400         3
2  CA-2016-138688  Office Supplies   14.6200         2
3  US-2015-108966        Furniture  957.5775         5
4  US-2015-108966  Office Supplies   22.3680         2


In [7]:
print("Before:", df.shape)
df.drop_duplicates(inplace=True)
print("After:", df.shape)

Before: (9994, 21)
After: (9994, 21)


In [8]:
df.to_csv('cleaned_superstore.csv', index=False)
print("Saved!")

Saved!


In [12]:
df['total_amount'] = df['Sales'] * df['Quantity']
print(df[['Sales', 'Quantity', 'total_amount']].head())
df.to_csv('cleaned_superstore.csv', index=False)
print("Saved!")

      Sales  Quantity  total_amount
0  261.9600         2      523.9200
1  731.9400         3     2195.8200
2   14.6200         2       29.2400
3  957.5775         5     4787.8875
4   22.3680         2       44.7360
Saved!


## Summary - Task 1: Superstore Dataset Exploration

- Dataset loaded: 9994 rows, 21 columns
- Missing values: None found in any column
- Duplicates removed: 0 duplicates found
- Filtered high sales (>1000): 468 rows
- New column added: total_amount = Sales * Quantity
- Cleaned file saved as cleaned_superstore.csv